applications/FluidDynamicsApplication/python_scripts/kratos_perturbation_process.py

```diff
@@ -0,0 +1,319 @@
+                node.SetValue(KratosMultiphysics.VELOCITY_LAPLACIAN, -1*node.GetSolutionStepValue(KratosMultiphysics.FluidDynamicsApplication.VELOCITY_NOISE)) # recall VELOCITY_NOISE is P(u_{cn}) (without penalty coefficient)
+
+    def FinalizeSolutionStep(self):
+        super(ImposePerturbedInitialConditionAnalysisStage,self).FinalizeSolutionStep()
+        KratosMultiphysics.ComputeNonHistoricalNodalGradientProcess(self._GetSolver().main_model_part,KratosMultiphysics.TEMPERATURE,KratosMultiphysics.VELOCITY_LAPLACIAN).Execute()
+        self.RestoreBoundaryValues()
+        # save new velocity
+        penalty_coeff = self.project_parameters["problem_data"]["penalty_coefficient"].GetDouble()
+        for node in self._GetSolver().main_model_part.Nodes:
+            velocity_field_loaded_and_correlated_noise = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY) # u_T + c*P(u_{cn})
+            gradient_solution = node.GetValue(KratosMultiphysics.VELOCITY_LAPLACIAN) # no-boundary: \nabla T ; boundary: - \nabla T
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY,velocity_field_loaded_and_correlated_noise + penalty_coeff*gradient_solution)
+
+
+class ImposePerturbedInitialConditionProcess(KratosMultiphysics.Process):
+    """
+    Process generating divergence-free correlated noise, imposing no-penetrability condition on bluff bodies,
+    preserving boundary conditions on boundaries and (if required) adding the contribution of a manually-loaded velocity field.
+    The generated initial field reads
+        u_0 = u_T + c*P(u_{cn}) + c*\nabla potential ,
+    where u_T is the velocity field we load, c a penalty coefficient, and P(u_{cn}) and \nabla potential are discussed next.
+    The process first projects the correlated noise u_{cn} into the model part. The projected correlated noise is denoted as P(u_{cn}).
+    Then, the associated poisson problem for the unknown TEMPERATURE is solved:
+        - \nabla \cdot \nabla TEMPERATURE = \nabla \cdot (c*P(u_{cn}) + u_{T})  on \Omega
+        c*P(u_{cn}) \cdot NORMAL = - \nabla TEMPERATURE \cdot NORMAL  on \partial \Omega_{bluff body}
+        TEMPERATURE = 0  on \Omega_{outlet}
+    Then, the tangential velocity is set to zero on the boundaries \partial \Omega_{bluff body} and \partial \Omega_{external domain} by doing:
+        P(u_{cn}) + \nabla TEMPERATURE = 0 on \Gamma
+    The correlated noise satisfying the no-penetrability and solenoidal conditions and respecting the boundary conditions is: VELOCITY = u_T + c*P(u_{cn}) + c*\nabla TEMPERATURE.
+
+    Parameters:
+    model    : Kratos Model
+    settings : Kratos Parameters
+    """
+
+    @property
+    def model_part_nodes(self):
+        return self.model_part.Nodes
+
+    def __init__(self,model,settings):
+        super().__init__()
+        for name, value in settings.items():
+            setattr(self, name, value)
+        self.model = model
```

Comment:
The settings here are read and set as python attributes. 

This should use the default_parameters mechanism.